src/js/node/events.js

```diff
@@ -386,21 +388,20 @@ async function* on(emitter, event, options = {}) {
 
   for (const evName of options?.close || []) {
     emitter.on(evName, () => {
-      emitter.emit(event, undefined);
       done = true;
     });
   }
 
-  while (!done) {
+  while (!done || !unconsumedEvents.isEmpty()) {
```

Comment:
Added this check to continue yielding the already read-in lines after the close event of `readline` is called. This fixes the issue that not all read-in lines were available in the `for await...of` loop.

Is there a situation in which the queued events should not be yielded and the loop should be stopped immediately after the close event is called?